src/stateManager/validators.ts

```diff
@@ -0,0 +1,328 @@
+
+/**
+ * Checks if enough time has passed (subjectively) to process this transaction
+ * Return SUCCESS if okay, or NOT_READY/DISPUTE/etc. if not.
+ */
+export const checkEnoughTimeSubjective = async <
+    TContext extends ValidationContext
+>({
+    stateManager,
+    signedBlock
+}: TContext): Promise<ValidationResult> => {
+    const flag =
+        await stateManager.isEnoughTimeToPlayMyTransactionSubjective(
+            signedBlock
+        );
+    return { executionFlag: flag };
+};
+
+/**
+ * Verifies that the block was produced by the correct participant
+ */
+export const checkCorrectBlockProducer = async <
+    TContext extends ValidationContext
+>({
+    stateManager,
+    block
+}: TContext): Promise<ValidationResult> => {
+    const nextToWrite = await stateManager.stateMachine.getNextToWrite();
+    if (block.transaction.header.participant !== nextToWrite) {
+        return {
+            executionFlag: ExecutionFlags.DISPUTE,
+            agreementFlag: AgreementFlag.INCORRECT_DATA
+        };
+    }
+    return { executionFlag: ExecutionFlags.SUCCESS };
+};
+
+//====================================================================
+// STATE TRANSITION VALIDATION
+//====================================================================
+
+/**
+ * Core function that verifies the transaction and updates state
+ * This function:
+ * 1. Captures current state hash
+ * 2. Applies the transaction
+ * 3. Verifies resulting state hash matches the block's stateHash
+ * 4. Adds the block to the manager if valid
+ */
+export const verifyStateTransition = async <
+    TContext extends ValidationContext
+>({
+    stateManager,
+    block,
+    signedBlock
+}: TContext): Promise<ValidationResult> => {
+    // Capture current state hash
+    const previousStateHash = await stateManager.getEncodedStateKecak256();
+
+    // Apply the transaction
+    const { success, encodedState, successCallback } =
+        await stateManager.applyTransaction(block.transaction);
+
+    // Compare resulting state hash with block's stateHash
+    const isStateHashValid =
+        ethers.keccak256(encodedState) === block.stateHash &&
+        previousStateHash === block.previousStateHash;
+
+    if (!success || !isStateHashValid) {
+        return {
+            executionFlag: ExecutionFlags.DISPUTE,
+            agreementFlag: AgreementFlag.INCORRECT_DATA
+        };
+    }
+
+    // Add the block to the manager
+    stateManager.agreementManager.addBlock(
+        block,
+        signedBlock.signature as SignatureLike,
+        encodedState
+    );
+
+    // Fire success callback asynchronously
+    setTimeout(() => {
+        if (!stateManager.isDisposed) {
+            successCallback();
+        }
+    }, 0);
+
+    return { executionFlag: ExecutionFlags.SUCCESS };
+};
+
+//====================================================================
+// CONFIRMATION VALIDATION FUNCTIONS
+//====================================================================
+
+/**
```

Comment:
This file needs to be split into BlockValidationPipeline and BlockConfirmationPipeline